sdk/php/runtime/main.go

```diff
@@ -0,0 +1,156 @@
+// Runtime module for the PHP SDK
+// when done add php to core/schema/sdk.go:L93
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"php-sdk/internal/dagger"
+)
+
+const (
+	PhpImage      = "php:8.3-cli-alpine@sha256:e4ffe0a17a6814009b5f0713a5444634a9c5b688ee34b8399e7d4f2db312c3b4"
+	ComposerImage = "composer:2@sha256:6d2b5386580c3ba67399c6ccfb50873146d68fcd7c31549f8802781559bed709"
+	EntryPoint    = "entrypoint.php"
+	ModSourceDir  = "/src"
+	GenDir        = "sdk"
+)
+
+type PhpSdk struct {
+	SourceDir     *dagger.Directory
+	RequiredPaths []string
+	Container     *dagger.Container
+}
+
+func New(
+	// Directory with the PHP SDK source code.
+	// +optional
+	sdkSourceDir *dagger.Directory,
+) *PhpSdk {
+	if sdkSourceDir == nil {
+		sdkSourceDir = dag.
+			CurrentModule().
+			Source().
+			Directory("../").
+			WithoutDirectory("runtime")
```

Comment:
Because of the difference between local and git remotes I think for now we need to duplicate the include/exclude patterns in a `WithDirectory` here. Hopefully when _context directories_ are available, the patterns will be able to be unified under a `// +ignore` pragma in this constructor's `sdkSourceDir` argument.

Reply:
https://github.com/dagger/dagger/blob/26ca9200d15dea7c43632c1c8b5ceb3e070db1c4/sdk/php/runtime/main.go#L38-L50

Reply:
What's in `sdkSourceDir` is what needs to go in the generated `sdk/` directory, generated for the module.

This is what I expect:

```go
// TODO: Replace with a *default path from context* when 
// https://github.com/dagger/dagger/pull/7744 becomes availble.
 sdkSourceDir = dag. 
 	Directory(). 
 	// NB: these patterns should match those in `dagger.json`. When `--sdk` points to a git remote
 	// the files aren't filtered using `dagger.json` include/exclude patterns since the whole repo is cloned.
 	// It's still useful to have the same patterns in `dagger.json` though, to avoid the unnecessary uploads
 	// when loading the SDK from a local path.
 	WithDirectory("", sdkRoot, dagger.DirectoryWithDirectoryOpts{
 	 	Include: []string{
			"composer.json",
			"src",
			"LICENSE",
			"README.md",
		},
 	 	Exclude: []string{
			"vendor",
		},
 	)
```

As I explained in other comments, I think you should move some files into `runtime/`, which is why the following files aren't in the list above:

- `template/` → `runtime/template/` (it's only used by the runtime to bootstrap a new module)
- `entrypoint.php`  → `runtime/scripts/entrypoint.php` (it's only as the runtime's entrypoint command)
- `init-template.sh` → `runtime/scripts/init-template.sh` (it's only by the runtime)
- ~`dagger.json`~  (not necesary as part of the SDK sources)
- ~`generated/`~ (codegen should be able to run with this being empty so it generates it from a clean slate)

For any of these, instead of getting them through `sdkSourceDir`, you should get them through the module's source directory. Remember that in `dagger.json` there's `"source": "runtime"`:
```go
tmplDir := dag.CurrentModule().Source().Directory("template")
entrypointFile := dag.CurrentModule().Source().File("scripts/entrypoint.php")
initTemplate := dag.CurrentModule().Source().File("scripts/init-template.sh")
```

Reply:
I've addressed this in my changes.